superset-frontend/src/explore/reducers/exploreReducer.js

```diff
@@ -131,11 +131,30 @@ export default function exploreReducer(state = {}, action) {
     },
     [actions.SET_FIELD_VALUE]() {
       const new_form_data = state.form_data;
+      const old_metrics_data = state.form_data.metrics;
+      const new_column_config = state.form_data.column_config;
```

Comment:
since we're modifying these two below, shouldn't we copy these two? Unless shallowly, like `{...state.form_data.column_config}`.

Reply:
@diegomedina248 , no need, because this is reducer section, and in reducer we return the state. So mutating old data has no effect to setting state again. There is no need to do shallow copy in reducer.